.github/workflows/deploy-server.yml

```diff
@@ -0,0 +1,71 @@
+name: Deploy to AWS
+
+on:
+  pull_request: # for now, trigger this on PR so I can test it before merging
+  push:
+    branches:
+      - master
+
+env:
+  AWS_REGION: us-east-2
+
+defaults:
+  run:
+    shell: bash
+
+jobs:
+  dockerimage:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+        name: Check out code
+
+      - uses: docker/build-push-action@v1
+        name: Build and Push Docker image
+        with:
+          repository: joshprzybyszewski/cribbage
+          username: ${{ secrets.DOCKER_USERNAME }}
+          password: ${{ secrets.DOCKER_PASSWORD }}
+          tag_with_ref: true
+          add_git_labels: true
+          tag_with_sha: false
+          cache_froms: joshprzybyszewski/cribbage:master
+
+  deploy:
+    needs: dockerimage
+    name: Deploy
+    runs-on: ubuntu-latest
+    steps:
```

Comment:
we'll want to find a way to only run this on master, but keep builds happening on PR

Reply:
Why do we want to build the docker image on every push to a PR?